devices/Lis2Mdl/Lis2Mdl.cs

```diff
@@ -0,0 +1,204 @@
+    {
+        /// <summary>
+        /// Configuration for enabling temperature compensation.
+        /// </summary>
+        private const byte ConfATemperatureCompensation = 0x80;
+
+        /// <summary>
+        /// Configuration for enabling block data read.
+        /// </summary>
+        private const byte ConfCBDU = 0x10;
+
+        /// <summary>
+        /// Device ID when reading the WHO_AM_I register.
+        /// </summary>
+        private const byte DeviceID = 0x40;
+
+        private I2cDevice _i2c;
+
+        /// <summary>
+        /// This holds the magnetic field reading from the sensor.
+        /// Elements index is 0 for X, 1 for Y and 2 for Z.
+        /// </summary>
+        private float[] _magneticFieldReadings = new float[3];
+
+        /// <summary>
+        /// Device I2C Address.
+        /// </summary>
+        public const byte I2cAddress = 0x1E;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Lis2Mdl" /> class.
+        /// </summary>
+        /// <param name="i2cDevice">I2C device.</param>
+        /// <exception cref="ArgumentNullException" >If the provided I2C device is <see langword="null"/>.</exception>
+        /// <exception cref="Exception">If the device is not found.</exception>
+        public Lis2Mdl(I2cDevice i2cDevice)
+        {
+            _i2c = i2cDevice ?? throw new ArgumentNullException(nameof(i2cDevice));
+
+            // check if the device is present
+            var id = Read(Register.WhoAmI);
+            if (id != DeviceID)
+            {
+                throw new Exception("Device not found");
+            }
+
+            InitializeRegisters();
+        }
+
+        /// <summary>
+        /// Temperature reading.
+        /// </summary>
+        [Telemetry]
+        public Temperature Temperature => Temperature.FromDegreesCelsius(GetTemperature());
+
+        /// <summary>
+        /// Gets the magnetic field reading for X axis.
+        /// </summary>
+        [Telemetry]
+        public MagneticField MagneticFieldX
+        {
+            get
+            {
+                GetMagneticField();
+
+                return UnitsNet.MagneticField.FromMilligausses(_magneticFieldReadings[0]);
+            }
+        }
+
+        /// <summary>
+        /// Gets the magnetic field reading for Y axis.
+        /// </summary>
+        [Telemetry]
+        public MagneticField MagneticFieldY
+        {
+            get
+            {
+                GetMagneticField();
+
+                return UnitsNet.MagneticField.FromMilligausses(_magneticFieldReadings[1]);
+            }
+        }
+
+        /// <summary>
+        /// Gets the magnetic field reading for Z axis.
+        /// </summary>
+        [Telemetry]
+        public MagneticField MagneticFieldZ
+        {
+            get
+            {
+                GetMagneticField();
+
+                return UnitsNet.MagneticField.FromMilligausses(_magneticFieldReadings[2]);
+            }
+        }
+
```

Comment:
For convenience, shall we add a Vector3 with the raw data in Milligauss in there?

Reply:
and "drop" the unit value? 🤔